### PR TITLE
Create palette

### PR DESCRIPTION
**Pull request by jessicalyn**
#### Description:
Save palette in redux and POST palette to database.

#### Where should the reviewer start?
start in Palettes.js

#### Link to related issues:
#4 

#### Covered by tests:
- [x] Yes
- [ ] No

#### if NO, explain why:
Need to update testing for: Palette.js, postNewPalette, actions, palettesReducer

#### Impacted Areas of Application:
Palettes

#### Todos before deployment:
- [ ] Update testing
- [ ] Save palette to specific project
- [ ] Fetch and store all palettes in redux state

#### Final checklist:
- [x] My code has no unused/commented out code
- [x] I have reviewed my code
- [x] I have commented my code, particularly in hard-to-understand areas

#### GIF of how this PR makes you feel:
Fun and creative GIF

### CODE REVIEW

File: palette-picker/src/thunks/postNewPalette.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+export const postNewPalette = (options, body) => {
+  return async (dispatch) => {
+    const url = `https://palette-picker-mfjk.herokuapp.com/api/v1/palettes`
+    try {
+      const response = await fetch(url, options)
+      if (!response.ok) {
+        throw Error (response.statusText)
+      }
+      const result = await response.json()
+      console.log("thunk", body)
```
Comment (kimmichurri) [inline palette-picker/src/thunks/postNewPalette.js]: remove console.log

File: palette-picker/src/reducers/palettesReducer.js [hunk 2]
```diff
@@ -0,0 +1,11 @@
+export const palettesReducer = (state = [], action) => {
+  switch(action.type) {
+    case 'STORE_PALETTES':
+      return action.palettes
+    case 'ADD_PALETTE':
+      console.log(action.newPalette)
```
Comment (kimmichurri) [inline palette-picker/src/reducers/palettesReducer.js]: remove console.log

File: palette-picker/src/containers/Palette/Palette.js [hunk 3]
```diff
@@ -17,12 +32,41 @@ export class Palette extends Component {
...
+      let colorKey = `color_${index + 1}`
+      this.setState({ [colorKey]: color.color })
+    })
+    this.addNewPalette()
+  }
+
+  addNewPalette = async () => {
+    const body = this.state
+    const options = await fetchOptionsCreator('POST', body)
+    await this.props.postNewPalette(options, body)
```
Comment (kimmichurri) [inline palette-picker/src/containers/Palette/Palette.js]: Do we need to await calling this thunk? We should have expect that the options creation is complete before this method is called.

File: palette-picker/src/containers/Color/Color.scss [hunk 4]
```diff
@@ -0,0 +1,14 @@
+.palette-color {
+  height: 100px;
+  margin: 0;
+  // display: flex;
```
Comment (kimmichurri) [inline palette-picker/src/containers/Color/Color.scss]: remove

File: palette-picker/src/containers/App/App.scss [hunk 5]
```diff
@@ -1,27 +1,53 @@
...
+  border: 2px solid orange;
+}
+
+.header-project-wrapper {
+  border: 2px solid blue;
+  flex: 1;
+}
+
+.app-header {
+  // height: 30%;
```
Comment (kimmichurri) [inline palette-picker/src/containers/App/App.scss]: remove

File: palette-picker/src/containers/Palette/Palette.js (hunk 3 above)
Comment (jessicalyn) [reply palette-picker/src/containers/Palette/Palette.js]: Let's test again when we pull down after the merge.

**Pull request merged**